lib/JsonNode.cpp

```diff
@@ -486,13 +486,235 @@ std::string JsonNode::toJson(bool compact) const
 
 ///JsonUtils
 
-void JsonUtils::parseTypedBonusShort(const JsonVector & source, const std::shared_ptr<Bonus> & dest)
+static void loadBonusSubtype(TBonusSubtype & subtype, BonusType type, const JsonNode & node)

```

Comment:
maybe move those large-parsing blocks from JsonNode.cpp to somewhere where bonus-related code is located?

Reply:
True, but probably not this time.
Our json parser files contain quite a lot of code largely not related to well, json parsing.
And not sure where such code should go. Bonus class itself? Bonus type handler? Or some separate entity? This code is used not just for bonus itself but also for limiters. There are also weird methods like "load town building bonus" one. Why that even a thing and not part of town handler?